Content.Server/Atmos/GasSprayerComponent.cs

```diff
@@ -0,0 +1,68 @@
+﻿using Content.Server.GameObjects.Components.Chemistry;
+using Content.Server.Interfaces;
+using Content.Shared.Chemistry;
+using Content.Shared.GameObjects.Components.Pointing;
+using Content.Shared.Interfaces.GameObjects.Components;
+using Robust.Server.GameObjects;
+using Robust.Server.GameObjects.EntitySystems;
+using Robust.Server.Interfaces.GameObjects;
+using Robust.Shared.GameObjects;
+using Robust.Shared.GameObjects.Systems;
+using Robust.Shared.IoC;
+using Robust.Shared.Localization;
+using Robust.Shared.Maths;
+using Robust.Shared.Serialization;
+
+
+namespace Content.Server.Atmos
+{
+    [RegisterComponent]
+    public class GasSprayerComponent : Component, IAfterInteract
+    {
+#pragma warning disable 649
+        [Dependency] private readonly IServerNotifyManager _notifyManager = default!;
+        [Dependency] private readonly IServerEntityManager _serverEntityManager = default!;
+#pragma warning restore 649
+
+        //TODO: create a function that can create a gas based on a solution mix
+        public override string Name => "GasSprayer";
+
+        private string _spraySound;
+        private string _sprayType;
+        private string _fuelType;
+
+        public override void ExposeData(ObjectSerializer serializer)
+        {
+            base.ExposeData(serializer);
+            serializer.DataField(ref _spraySound, "spraySound", string.Empty);
+            serializer.DataField(ref _sprayType, "sprayType", string.Empty);
+            serializer.DataField(ref _fuelType, "fuelType", string.Empty);
+
+        }
+
+        public void AfterInteract(AfterInteractEventArgs eventArgs)
+        {
+            if (Owner.TryGetComponent(out SolutionComponent tank) &&
+                tank.Solution.GetReagentQuantity(_fuelType).Float().Equals(0f))
+            {
+                _notifyManager.PopupMessage(Owner, eventArgs.User,
+                    Loc.GetString("The {0} is out of {1}!", Owner.Name, _fuelType));
```

Comment:
Maybe get a better name for the fueltype? "The fire extinguisher is out of chem.H2O" doesn't sound that nice